### DISCUSSION TITLE
Do you also publish the HMI / TFT file for the Nextion?

### DISCUSSION THREAD

**Discussion by ghost** [GillesLACAUD/OPLA-Source-code#27] [General]
Do you also publish the HMI / TFT file for the Nextion?

**Comment by GillesLACAUD** [comment]
yes i will....i just need to find the time to do it.
There is another github repo for these documents

**Comment by GillesLACAUD** [comment]
OPLA other files.
Please check this repo

**Comment by ghost** [comment]
Thank you very much, I can currently find the Hardware/PCBs there, I'll have a look again.

**Comment by GillesLACAUD** [comment]
Nextion files are upload

**Comment by ghost** [comment]
Thank you very much, would you please also add the files for the SD-Card (AKWF, sound, System)?

**Comment by GillesLACAUD** [comment]
yes....of course....i will add all the informations... :-)

**Reply by Electrotechnique** [reply]
Hello, I saw OPLA and decided to build it. because I have a Nextion Display and ESP32-A1s already. The Nextion is a NX4827T043, which is bigger but I have recompiled the code for it and it seems to work. Are you powering the display from 3V3 on the ESP32 board? Because it requires 5V to work. 

Thanks very much for making this open source.

**Reply by ghost** [reply]
Hello, 
I use 5V (VCC5V) at D1/C3.

**Comment by GillesLACAUD** [comment]
Hello Electro technique and Wlad
The schematic is here.....the connector for the Nextion is the PL1 (check the pdf files)

https://github.com/GillesLACAUD/OPLA-OTHER-FILES/tree/main/01%20Hardware/V2.0%20Minimal%20Version/Plots

You can find some usefull informations in the OPLA-OTHER-FILES github

Add please check witch version of the A1S do you have.
- The version with an AC101 codec work fine but the audio in feature is not present
- There is 2 versions of the A1S with the ES8388 codec
- When the OPLA start it will print the codec and his version on the nextion screen.
- If you have an ES8388V1 version the OPLA code do not work fine (the mapping for this version is not correct)

Good luck for your OPLA....
Please send us a Photo when you have finish... :-)

**Reply by Electrotechnique** [reply]
Hello, I've looked at the gerbers. Are the PCB files Eagle? I just want to look at the schematic. 

-- Ok, I think schematics are on PDFs. You appear to be using 3V3 to power the display?

**Reply by Electrotechnique** [reply]
Ok, I've got it going. I connected the display to 5V and used a voltage divider to bring down the TX from the display to 3V3. I had a MIDI input already made because I was playing with Marcel's ESP32 synths.

It sounds promising and quite capable.
![20220311_163850](https://user-images.githubusercontent.com/50934180/157824627-f68edf9e-a856-433d-85f1-4da05f3c016d.jpg)

**Comment by GillesLACAUD** [comment]
Hi ElectroTechnique
Thank you to share with us the photo :-)
Maybe you can find a solution for the polyphonie...the OPLA had only 4 voices and the Marcel Licence 6 or 8 voices, and I do not understand why. I have also some issue with the delay and I do not understand too....
Of course if you have any question or idea for future features...

Did you plan to modify the code ?

Musically
Hansy Synth

**Reply by ghost** [reply]
In my opinion, the use of psram greatly reduces the speed of program execution. the psram is connected externally to the esp32 and is not as fast as the internal one.
If i understand correctly, marcel doesn't use the psram for the synths and thus achieves a higher number of voices. For reverb and delay he uses the little remaining internal memory.

**Reply by GillesLACAUD** [reply]
yes it makes sense...I have check the new code of ML...but even if I comment the delay I can not have a poly above 4.

**Reply by ghost** [reply]
I think you have to compile the code without the option 'use psram' and don't use 'ps_malloc' in your code

**Reply by GillesLACAUD** [reply]
Ok I will try :-)
